File: Leetcode-Practice/Datastructure/src/ds/stack/stack.java
package ds.stack;

public class stack {

	int stack[] = new int[5];
	int top =0;

	
	public void push(int data) {
		stack[top]=data;
		top++;
	}
	
	public int pop() {
		
		int data=0;
		if(isEmpty()) {
			System.out.println("Stack is empty");
		}else {
			
			top--;
			data = stack[top];
			stack[top]=0;
		}
		
		return data;
	}
	
	public int peek() {
		int data;
	
		data = stack[top -1];
		stack[top]=0;
		return data;
		
	}
	
	public int size() {
//		int size=0;
//		for (int n : stack) {
//			size++;
//		}
		return top;
	}
	
	public boolean isEmpty() {
		
		return top<=0;
	}
	public void show() {
		for (int n : stack) {
			System.out.println(n+ " ");
		}
	}
}

File: Leetcode-Practice/stack.js

class Node{

    constructor(value){
        this.value=value;
        this.next=null;
    }
}


class Stack {

    constructor(){

        this.top = null;
        this.bottom = null;
        this.length = 0;
    }

    peek(){
        return this.top;

    }

    push(value) {

        const newNode = new Node(value);

        if (this.length==0){
            this.top=newNode;
            this.bottom=newNode;
        }else{
            const holdingpointer = this.top;
            this.top=newNode;
            this.top.next = holdingpointer;
        }
        this.length++;
        return this;

    }

    pop(){

        if (!this.top){
            return null;
        }

        if (this.top=== this.bottom){
            this.bottom=null;
        }
        const holdingpointer = this.top;
        this.top=this.top.next;
        this.length--;
        return this;
    }
}

const myStack = new Stack();
myStack.push('aninda');
myStack.push('mukherjee');
myStack.push('Jon');
myStack.pop();

console.log(myStack.peek());